src/Visualization/Python/Render1D.py

```diff
@@ -0,0 +1,279 @@
+import sys
+import os
+import numpy as np
+import logging
+import matplotlib as mpl
+try:
+    import Tkinter as tkinter
+except ImportError:
+    import tkinter
+
+
+def find_extrema_over_data_set(arr):
+    '''
+    Find max and min over a range of number arrays
+    :param arr: the array over which to find the max and min
+    '''
+
+    return (np.nanmin(arr), np.nanmax(arr))
+
+
+def parse_cmd_line():
+    '''
+    parse command-line arguments
+    :return: dictionary of the command-line args, dashes are underscores
+    '''
+
+    parser = argparse.ArgumentParser(
+        description='Render 1-dimensional data',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    group_files = parser.add_mutually_exclusive_group(required=True)
+    group_files.add_argument(
+        '--file-prefix',
+        type=str,
+        help='Common prefix of all h5 files being used '
+        'in the case of a single h5 file, it is the file name '
+        'without the .h5 extension')
+    group_files.add_argument('--filename-list',
+                             type=str,
+                             nargs='+',
+                             help='List of all files if they do not '
+                             'have same file prefix. You must include \'.h5\' '
+                             'extension ')
+    group_vars = parser.add_mutually_exclusive_group(required=True)
+    group_vars.add_argument('--var',
+                            type=str,
+                            help='Name of variable to render. E.g. \'psi\' '
+                            'or \'Error(psi)\'')
+    group_vars.add_argument(
+        '--list-vars',
+        type=bool,
+        help='If true then print to screen vars in h5 file '
+        'and exit')
+    parser.add_argument(
+        '--time',
+        type=int,
+        required=False,
+        help='If specified renders the integer observation step'
+        ' instead of an animation')
+    parser.add_argument('-o',
+                        '--output',
+                        type=str,
+                        required=False,
+                        help='Set to the name of the output file you want '
+                        'written. For animations this saves an mp4 file and '
+                        'for stills a pdf. Name of file should not include '
+                        'file extension. If not specified, renders to screen.')
+    group_anim = parser.add_mutually_exclusive_group(required=False)
+    group_anim.add_argument(
+        '--fps',
+        type=float,
+        default=5,
+        help='Set the number of frames per second when writing '
+        'an animation to disk.')
+    group_anim.add_argument('--interval',
+                            type=float,
+                            help='Delay between frames in  milliseconds')
+    # Check for h5 suffix if '--file-prefix' argument used
+    args = parser.parse_args()
+    assert args.file_prefix is None or not args.file_prefix.endswith('.h5'),\
```

Comment:
Another FYI (not a change request): It can be useful to wrap code such as the condition of this assert or the string in parentheses `()`. Then they can span multiple lines, you don't need the `\` and `yapf` will format it automatically.